File: C++/Anfänger/StatCalc.cpp
#include <iostream> // pre-prozessor command 
#include <cmath> // pre-prozessor command 
using namespace std; // declare namespace

class StatCalc {

private:
 int count;   // Number of numbers that have been entered.
 double sum;  // The sum of all the items that have been entered.
 double squareSum;  // The sum of the squares of all the items.
 double max, min; 

public:
  StatCalc() : count(0), sum(0.), squareSum(0.), max(0.), min(0.) {} // default constructor

  void enter(double num) {
    // Add the number to the dataset.
    if (count == 0)
	min=max=num;
    count++;
    sum += num;
    squareSum += num*num;
    if (num < min)   
	min = num;
    if (num > max)
	max = num;
  }

  int getCount() {   
    // Return number of items that have been entered.
    return count;
  }
  
  double getSum() {
    // Return the sum of all the items that have been entered.
    return sum;
  }

  double getMean() {
    // Return average of all the items that have been entered.
    // Value is Double.NaN if count == 0.
    return sum / count;  
  }
  
  double getStandardDeviation() {  
    // Return standard deviation of all the items that have been entered.
    // Value will be Double.NaN if count == 0.
    double mean = getMean();
    return sqrt( squareSum/count - mean*mean );
  }

  double getMin() {  
    return min;
  }

  double getMax() {  
    return max;
  }

};  // end of class StatCalc

#include <cstdlib>
#include <time.h>

int main() 
{

  StatCalc s1;  // lege StatCalc Objekt an

  srandom(time(0)); // setup random generator

  for ( int i=0; i<1000; i++ ) {
    double xrnd = random()*2./RAND_MAX - 1.; // Zufallszahlen zwischen [-1,1]
    s1.enter(xrnd);
  }

  cout << s1.getCount() << "   "
       << s1.getMean()  << "   "
       << s1.getStandardDeviation()  << "   "
       << s1.getMax()  << "   "
       << s1.getMin()  << "   "
       << endl;




  return(0);
}

File: Python/makini/Python/StatCalc.py
#!/usr/bin/env python

import math
#
#   An object of class StatCalc can be used to compute several simple statistics
#   for a set of numbers.  Numbers are entered into the dataset using
#   the enter(double) method.  Methods are provided to return the following
#   statistics for the set of numbers that have been entered: The number
#   of items, the sum of the items, the average, and the standard deviation.
#

class StatCalc(object):
		

	def __init__(self):
		self.count = 0
		self.sum   = 0.
		self.squareSum   = 0.
		self.werte = []

	def enter(self, num):
		" Add the number to the dataset."
		self.count += 1
		self.sum += num
		self.squareSum += num*num
		self.werte.append(num)


	def getCount(self): 
		" Return number of items that have been entered."
		return self.count

	def getSum(self): 
		if self.count == 0:
			return "Keine Werte Enthalten"
		" Return the sum of all the items that have been entered."
		return self.sum

	def getMean(self): 
		" Return average of all the items that have been entered."
		if self.count == 0:
			return "Keine Werte Enthalten"
		return self.sum / self.count

	def getStandardDeviation(self): 
		" Return standard deviation of all the items that have been entered."
		if self.count == 0:
			return "Keine Werte Enthalten"
		return math.sqrt( self.squareSum/self.count - self.getMean()**2 )
	
	def getMinimum(self):
		try:
			min = self.werte[0]
			for i in range(self.getCount()):
				if min > self.werte[i]:
					min = self.werte[i]
			return min
		except:
			print "Keine Werte"

	def getMaximum(self):
		try:
			max = self.werte[0]
			for i in range(self.getCount()):
				if max < self.werte[i]:
					max = self.werte[i]
			return max
		except:
			print "Keine Werte"
   
# end of class StatCalc